tests/UI/Component/Chart/Speedo/ChartSpeedoTest.php

```diff
@@ -0,0 +1,209 @@
+        ));
+
+        $this->assertInstanceOf("ILIAS\\UI\\Component\\Chart\\Speedo\\Standard", $responsive);
+
+        $html = $r->render($responsive);
+
+        $expected_html =
+            '<div class="il-chart-speedo-box responsive">' .
+            '  <div class="il-chart-speedo-container">' .
+            '    <div class="il-chart-speedo-marker-box" style="transform: rotate(57.5deg)">' .
+            '      <div class="il-chart-speedo-marker"><div class="il-chart-speedo-arrow"></div></div>' .
+            '    </div>' .
+            '    <div class="il-chart-speedo-outerbox il-chart-speedo-red">' .
+            '      <div class="il-chart-speedo-pointer pointer-1" style="transform: rotate(-25deg) skew(41.6349deg)"></div>' .
+            '      <div class="il-chart-speedo-pointer pointer-2" style="transform: rotate(23.3651deg) skew(41.6349deg)"></div>' .
+            '      <div class="il-chart-speedo-pointer pointer-3" style="transform: rotate(71.7302deg) skew(41.6349deg)"></div>' .
+            '      <div class="il-chart-speedo-cover">' .
+            '        <div class="il-chart-speedo-text-container">' .
+            '          <span class="il-chart-speedo-score-text"></span>' .
+            '          <span class="il-chart-speedo-score">63 %</span>' .
+            '          <span class="il-chart-speedo-minscore">75 %</span>' .
+            '          <span class="il-chart-speedo-minscore-text"></span>' .
+            '        </div>' .
+            '      </div>' .
+            '    </div>' .
+            '  </div>' .
+            '</div>';
+
+        $this->assertHTMLEquals($expected_html, $html);
+    }
+
+    public function test_render_mini()
+    {
+        $r = $this->getDefaultRenderer();
+        $f = new \ILIAS\UI\Implementation\Factory();
+        $mini = $f->speedo()->mini(array(
+            'goal' => 400,
+            'score' => 250,
+            'minimum' => 300,
+        ));
+
+        $this->assertInstanceOf("ILIAS\\UI\\Component\\Chart\\Speedo\\Mini", $mini);
+
+        $html = $r->render($mini);
+
+        $expected_html =
+            '<div class="il-chart-speedo-box mini">' .
+            '  <div class="il-chart-speedo-mini-container">' .
+            '    <div class="il-chart-speedo-outerbox il-chart-speedo-red">' .
+            '      <div class="il-chart-speedo-pointer pointer-1" style="transform: rotate(-25deg) skew(41.6349deg)"></div>' .
+            '      <div class="il-chart-speedo-pointer pointer-2" style="transform: rotate(23.3651deg) skew(41.6349deg)"></div>' .
+            '      <div class="il-chart-speedo-pointer pointer-3" style="transform: rotate(71.7302deg) skew(41.6349deg)"></div>' .
+            '      <div class="il-chart-speedo-cover">' .
+            '        <div class="il-chart-speedo-text-container"></div>' .
+            '      </div>' .
+            '    </div>' .
+            '  </div>' .
+            '</div>';
+
+        $this->assertHTMLEquals($expected_html, $html);
+    }

```

Comment:
What about tests for those "get as percentage" like behaviours.

Reply:
I think you are right and I should test these either. I will implement this tests as soon as I add the new ones. Currently, the tests will fail anyway, because they have not yet been switched to the new paths.
(They currently call "$f->speedo()" instead of "$f->chart()->speedo()", because I have written and tested them before I moved the component to the chart family)